Src/Support/Google.Apis/Services/BaseClientService.cs

```diff
@@ -236,6 +261,39 @@ protected virtual BackOffHandler CreateBackOffHandler()
             return new BackOffHandler(new ExponentialBackOff());
         }
 
+        /// <summary>
+        /// Gets the effective URI taking into account the <see cref="UniverseDomain"/>.
+        /// </summary>
+        /// <param name="explicitUri">An explicit URI. May be null.</param>
+        /// <param name="defaultUri">A default URI. May be null.</param>
+        /// <returns>
+        /// <list type="bullet">
+        /// <item><paramref name="explicitUri"/> if not null.</item>
+        /// <item>
+        /// Otherwise, if <see cref="UniverseDomain"/> is not null, the result of replacing
+        /// <see cref="DefaultUniverseDomain"/> with <see cref="UniverseDomain"/>
+        /// in <paramref name="defaultUri"/>.
+        /// </item>
+        /// Otherwise <paramref name="defaultUri"/>.
+        /// </list>
+        /// </returns>
+        protected internal string GetEffectiveUri(string explicitUri, string defaultUri)
+        {
+            // Note this method needs to receive explicit and default URIs so we can use
+            // it for the batch endpoint as well. The batch endpoint does not have an
+            // override mechanism, so we pass explicitUri as null in that case.
+
+            if (explicitUri is not null)
```

Comment:
Possibly convert this all into a conditional ?: operator? (Could even use ?? and then a single ?: operator:

```csharp
explicitUri ??
(UniverseDomain is not null 
? defaultUri?.Replace(DefaultUniverseDomain, UniverseDomain)
: defaultUri)
```